vulnerabilities/pipelines/npm_importer.py

```diff
@@ -58,9 +73,26 @@ def advisories_count(self):
 
     def collect_advisories(self) -> Iterable[AdvisoryData]:
         vuln_directory = Path(self.vcs_response.dest_dir) / "vuln" / "npm"
-
-        for advisory in vuln_directory.glob("*.json"):
-            yield from self.to_advisory_data(advisory)
+        advisory_files = list(vuln_directory.glob("*.json"))
+
+        if not self.is_batch_run:
+            package_name = self.purl.name
```

Comment:
Why we not actually check in our DB if the purl exists, before even checking all files ?